key-attestation/java-example/src/main/java/com/fortanix/keyattestationstatementverifier/Verify.java

```diff
@@ -0,0 +1,306 @@
+            Object object = pemParser.readObject();
+            while (object != null) {
+                if (object instanceof X509CertificateHolder) {
+                    X509CertificateHolder cert = (X509CertificateHolder) object;
+                    if (!certChain.add(cert)) {
+                        throw new KeyAttestationStatementVerifyException("Failed to add new Certificate");
+                    }
+                } else {
+                    throw new KeyAttestationStatementVerifyException(
+                            "Found a non Certificate PEM object: " + object.toString());
+                }
+                object = pemParser.readObject();
+            }
+            pemParser.close();
+        }
+        if (certChain.isEmpty()) {
+            throw new KeyAttestationStatementVerifyException("No valid Certificate in given reader");
+        }
+        return convertX509CertificateHolders(certChain);
+    }
+
+    /**
+     * Helper function to convert a List of
+     * org.bouncycastle.cert.X509CertificateHolder to a List of
+     * java.security.cert.X509Certificate
+     *
+     * @param cert_chain List of certificate to be converted
+     * @return A List of java.security.cert.X509Certificate
+     * @throws Exception
+     */
+    public static List<X509Certificate> convertX509CertificateHolders(List<X509CertificateHolder> cert_chain)
+            throws Exception {
+        LOGGER.info("Converting List<X509CertificateHolder> to List<X509Certificate> ...");
+        // Convert X509CertificateHolder to Certificate for CertPath
+        JcaX509CertificateConverter converter = new JcaX509CertificateConverter()
+                .setProvider(BouncyCastleProvider.PROVIDER_NAME);
+        List<X509Certificate> chain = new ArrayList<>();
+        for (X509CertificateHolder holder : cert_chain) {
+            X509Certificate cert = converter.getCertificate(holder);
+            LOGGER.fine("Converted cert:\n" + cert.toString());
+            chain.add(cert);
+        }
+        return chain;
+    }
+
+    /**
+     * Helper function to check the length of authority certificate chain.
+     *
+     * @param authorityCertChain Certificate chain to be checked
+     * @throws KeyAttestationStatementVerifyException
+     */
+    private static void checkAuthorityChainLength(List<X509Certificate> authorityCertChain)
+            throws KeyAttestationStatementVerifyException {
+        int authorityCertChainSize = authorityCertChain.size();
+        LOGGER.info(String.format("Checking authorityCertChain size: %d == %d ?", authorityCertChainSize,
+                Common.VALID_AUTHORITY_CERT_CHAIN_NUM));
+        if (authorityCertChainSize != Common.VALID_AUTHORITY_CERT_CHAIN_NUM) {
+            throw new KeyAttestationStatementVerifyException(
+                    String.format(
+                            "A valid authority certificates chain should contain %d certificate",
```

Comment:
Since I assume the `authorityCertChain` comes from backend, so I also assume the length is three at current time point. We could update this part of code in future if we have any change.

Reply:
That's fair, I guess. Resolving for now.